design-document.md

```diff
@@ -14,4 +24,171 @@
 
 ## Open questions and system parameters
 
-## Cryptographic dependencies
+## Cryptographic components
+
+### Overview of the core protocol
+
+**Notation.**
+Fix a finite field K. We write the linear, s-way secret sharing of an element x
+of K^n as {x:1}, ..., {x:s}, where {x:i} is the share held by the i-th party. We
+write {x} as shorthand for the sequence {x:1}, ..., {x:s}.
+
+Prio combines standard linear secret sharing with a new type of
+probabilistically checkable proof (PCP) system, called a fully linear PCP. The
+core protocol can be described in terms of three main algorithms:
+
+1. pf := Prove(x) denotes generation of a proof pf of the validity of input x.
+   This algorithm is executed by the user.
+1. {vf:i} := Query({x:i}, {pf:i}, r) denotes computation of the verification
+   share {vf:i} for input share {x:i} and proof share {pf:i}. This algorithm is
+   executed by each of the aggregators; input r denotes the joint randomness
+   shared by all of the aggregators.
+1. b := Decide({vf}, r) denotes the execution of the decision procedure on input
+   shares {vf} and joint randomness r. The output b is a boolean indicating
+   whether the input is deemed valid. This algorithm is run by the leader.
+
+The values above have following types:
+
+1. Input x is an element of K^n for some integer n.
+1. Proof pf is an element of K^p(n) for some function p.
+1. The joint randomness r is an element of K^u(n) for some function u.
+1. Each verification share {vf:i} is an element of K^v(n) for some function v.
```

Comment:
Out of curiosity, why use separate functions to denote the extension of K for each type? Is it not the case that `n = p(n) = u(n) = v(n)`?

Reply:
Actually "K^n" is meant to denote the set of vectors over K of length n. :grimacing: I added this notation to the "Notation" paragraph above.

To your question: in general, the length of the proof, verification message, and joint randomness is a function of the length of length of the input. For example, for an input length of n, the corresponding proof might be p(n) = a*n + b for some constants a, b.